File: forex.insights.spa/src/api/feedback/interfaces/feedback-post-request.ts
/**
 * Interface for the feedback post request.
 */
interface FeedbackPostRequest {
    /**
     * First name.
     */
    firstName: string;

    /**
     * Last name.
     */
    lastName: string;

    /**
     * User's feedback.
     */
    feedback: string;
}

export type { FeedbackPostRequest };

File: forex.insights.api/DataModels/Feedbacks/FeedbackPostRequest.cs
using System.ComponentModel.DataAnnotations;

namespace forex.insights.api.DataModels.Feedbacks
{
    /// <summary>
    /// Feedback post request.
    /// </summary>
    public class FeedbackPostRequest
    {
        /// <summary>
        /// First name.
        /// </summary>
        [Required]
        [MaxLength(20)]
        public string FirstName { get; set; } = string.Empty;

        /// <summary>
        /// Last name.
        /// </summary>
        [Required]
        [MaxLength(20)]
        public string LastName { get; set; } = string.Empty;

        /// <summary>
        /// Feedback.
        /// </summary>
        [Required]
        [MaxLength(250)]
        public string Feedback { get; set; } = string.Empty;
    }
}
